src/ApplicationDefaultCredentials.php

```diff
@@ -321,6 +328,40 @@ public static function getIdTokenCredentials(
         return $creds;
     }
 
+    /**
+     * Returns a StdOutLogger instance
+     *
+     * @return null|LoggerInterface
+     */
+    public static function getDefaultLogger(): null|LoggerInterface
+    {
+        $loggingFlag = getenv(self::SDK_DEBUG_FLAG);
+
+        // Env var is not set
+        if (!is_string($loggingFlag)) {
+            if (is_array($loggingFlag)) {
+                trigger_error('The ' . self::SDK_DEBUG_FLAG . ' is set, but it is set to another value than false, true, 0 or 1. Logging is disabled');
+                return null;
+            }
+
+            return null;
+        }
```

Comment:
I remember having a specific issue that I needed to handle but I can't quite remember what was it exactly. Although it might have been related to PHPSTAN mentioning that it can return an `array`? But I am unsure. 

Still let me take a look in order to make it cleaner :)